yarn-project/noir-libs/value-note/src/filter.nr

```diff
@@ -6,6 +6,10 @@ fn get_2_notes<P>(notes: [Option<ValueNote>; MAX_READ_REQUESTS_PER_CALL], _x: P)
     [notes[0], notes[1]]
 }
 
+fn get_4_notes<P>(notes: [Option<ValueNote>; MAX_READ_REQUESTS_PER_CALL], _x: P) -> [Option<ValueNote>; 4] {
```

Comment:
@sirasistant are we still unable to have a generic get_notes function? (generic over the number of notes)?
(Also, what's `P`?)

Reply:
It should be possible, using slices

```
fn get_p_notes<P>(notes: [Option<ValueNote>; MAX_READ_REQUESTS_PER_CALL], p: Field) ->  [Option<ValueNote>]
```
or changing the signature to pass an empty array of the correct size
```
fn get_p_notes<P>(notes: [Option<ValueNote>; MAX_READ_REQUESTS_PER_CALL], p: [Option<ValueNote>; P]) ->  [Option<ValueNote>; P]
```